sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsController.java

```diff
@@ -208,15 +210,34 @@ public void deleteEventParticipant(String eventUuid, String personUuid, Runnable
 				EventParticipantDto dto = editForm.getValue();
 				EventDto eventDto = FacadeProvider.getEventFacade().getEventByUuid(dto.getEvent().getUuid());
 				UserDto user = UserProvider.getCurrent().getUser();
-				if ((user.getRegion() != null && dto.getRegion() != null && !user.getRegion().equals(dto.getRegion()))
-					|| (user.getDistrict() != null && dto.getDistrict() != null && !user.getDistrict().equals(dto.getDistrict()))
-					|| (dto.getRegion() == null
-						&& dto.getDistrict() == null
-						&& (user.getRegion() != null && !user.getRegion().equals(eventDto.getEventLocation().getRegion())
-							|| user.getDistrict() != null && !user.getDistrict().equals(eventDto.getEventLocation().getDistrict())))
-					|| ((dto.getRegion() == null || dto.getDistrict() == null)
-						&& (user.getRegion() != null && !user.getRegion().equals(dto.getRegion())
-							|| user.getDistrict() != null && !user.getDistrict().equals(dto.getDistrict())))) {
+
+				RegionReferenceDto userRegion = user.getRegion();
+				DistrictReferenceDto userDistrict = user.getDistrict();
+
+				RegionReferenceDto epResponsibleRegion = dto.getRegion();
+				DistrictReferenceDto epResponsibleDistrict = dto.getDistrict();
+
+				RegionReferenceDto epEventRegion = eventDto.getEventLocation().getRegion();
+				DistrictReferenceDto epEventDistrict = eventDto.getEventLocation().getDistrict();
+
+				Boolean responsibleRegionDiffersFromUserRegion =
```

Comment:
I would chose better naming for the boolean variables

Reply:
I've made some changes but I could not get more inspired than that. I also added comments explaining each boolean variable.